include/range/v3/utility/tagged_pair.hpp

```diff
@@ -14,175 +15,196 @@
+                    noexcept(std::is_nothrow_move_constructible<Untagged>::value)    \
+                  : Next(detail::move(that))                                         \
+                {}                                                                   \
+                CONCEPT_REQUIRES(CopyConstructible<Untagged>())                      \
+                constexpr getter(Untagged const &that)                               \
+                    noexcept(std::is_nothrow_copy_constructible<Untagged>::value)    \
+                  : Next(that)                                                       \
+                {}                                                                   \
+                getter &operator=(getter &&) = default;                              \
+                getter &operator=(getter const &) = default;                         \
+                RANGES_CXX14_CONSTEXPR                                               \
+                meta::_t<std::tuple_element<I, Untagged>> &NAME() &                  \
+                    noexcept(noexcept(                                               \
+                        detail::adl_get<I>(std::declval<Untagged &>())))             \
+                {                                                                    \
+                    return detail::adl_get<I>(static_cast<Untagged &>(*this));       \
+                }                                                                    \
+                RANGES_CXX14_CONSTEXPR                                               \
+                meta::_t<std::tuple_element<I, Untagged>> &&NAME() &&                \
+                    noexcept(noexcept(                                               \
+                        detail::adl_get<I>(std::declval<Untagged>())))               \
+                {                                                                    \
+                    return detail::adl_get<I>(static_cast<Untagged &&>(*this));      \
+                }                                                                    \
+                constexpr                                                            \
+                meta::_t<std::tuple_element<I, Untagged>> const &NAME() const &      \
+                    noexcept(noexcept(                                               \
+                        detail::adl_get<I>(std::declval<Untagged const &>())))       \
+                {                                                                    \
+                    return detail::adl_get<I>(static_cast<Untagged const &>(*this)); \
+                }                                                                    \
+            };                                                                       \
+        };                                                                           \
+    }                                                                                \
```

Comment:
Primarily:
* Use ADL for `get` so that non-standard types are taggable
* Inject the tagged getters between the top-level tagged type and the base type, instead of using sibling inheritance, so that tagged getters hide members of the same name inherited from the base instead of being ambiguous. (This wreaked all kinds of havoc with recursive composition of tagged types, which became fairly common with the `compressed_pair` change.)